README.md

```diff
@@ -1,63 +1,48 @@
+This project demonstrates how to run a basic ROS2 end-to-end test in simulation (here turtlesim).
 
-## Prerequisite
-
-You will need ROS2 galactic or humble installed.
-
-See instructions [here](https://docs.ros.org/en/humble/Installation.html)
-
-## ROS2 tests
-
-You can run the tests with:
-
-```
-launch_test launch_turtle.py
-```
+This demo runs either natively or in containers. Native runs work only on Ubuntu version compatible with the tested ROS2 Galactic and Humble versions. Container runs work on Linux, Darwin and Windows, with the later run in X11-compatible shells.
 
-As it is, the test will fail.
+## Prerequisite
 
-If you take a look at the code, you should be able to find what is wrong and make the test pass.
+### Container for Linux/Darwin/Windows
 
+* Container system: Docker (Podman coming)
+* X11 window system, like XQuartz on Darwin and MobaXterm on Windows, configured to accept local network connections.
 
-## Tests with ArtefactsCI
+### Native for supported Ubuntu
 
+This is tested and likely to work only on ROS2 Galactic/Humble Ubuntu systems.
 
-You can use one of the 2 options below.
+See instructions for example [here for Humble](https://docs.ros.org/en/humble/Installation.html)
 
-### With local ROS2 install
+## Usage
 
-Additionally, you can use ArtefactsCI to keep track of your tests results.
+### Native for supported Ubuntu
 
-1. Create an account and project on ArtefactsCI
+#### Preliminary: Direct test run
 
-2. Install the artefacts client
+You can run the tests directly with:
 
 ```
-pip install artefacts-client --extra-index-url https://d5cw4z7oemmfd.cloudfront.net/pep503/ -U
+launch_test launch_turtle.py
 ```
 
-2. Edit the project name in artefacts.yaml
-
-3. Run the tests with tracking enabled
-
-```
-artefacts run basic_tests
-```
+The test should succeed. One way to make it fail for demo purpose is to, say, increment by 0 the turtle velocity, so it never reaches the edge expeced by the test case (see line 19 in `sample_node.py`).
 
-### Using Docker
+#### Tests with Artefacts
 
+1. Create a free account and project on [Artefacts](https://app.artefacts.com)
+2. Install the Artefacts CLI: `pip3 install artefacts-cli`
+3. Edit the project name in `artefacts.yaml` to the name of your project in Artefacts (step 1).
+4. Run the tests with tracking enabled: `artefacts run basic_tests`
+5. Results and detail should shortly be available in your Artefacts dashboard.
 
-Tests on MacOS and Ubuntu 20 often require explictly authorizing access to X11:
+### Container runs
 
-    [MacOS] xhost +127.0.0.1: This will start an X11 server if needed, and allow the container to connect.
-    [Ubuntu 20] xhost +local:docker
+Runs should work out of the box, assuming complete X11 configuration:
 
-Then build and run the docker container:
+    artefacts run --in-package basic_tests
 
-```
-git clone git@github.com:art-e-fact/warp-client.git
```

Comment:
maybe we should keep the menton that you need to clone before doing the artefacts run?

Reply:
I have added on cloning under the Usage section. Does that flow well for now?